File: guanabara-java/src/com/danillo/algoritmos/lista04/Exercicio53.java
package com.danillo.algoritmos.lista04;

import java.util.Scanner;

public class Exercicio53 {
    public static void main(String[] args) {
        Scanner entrada = new Scanner(System.in);

        int i = 0, mulheresCadastradas = 0, homensCadastrados = 0, mulheresMais20 = 0, soma = 0, somaHomens = 0;

        while(i < 5){
            System.out.println("Insira o gênero (M ou F): ");
            char genero = entrada.next().charAt(0);
            System.out.println("Insira a idade: ");
            int idade = entrada.nextInt();

            if(genero == 'F'){
                mulheresCadastradas++;
            }else{
                homensCadastrados++;
            }

            if(genero == 'F' && idade > 20){
                mulheresMais20++;
            }

            soma = soma + idade;

            if(genero == 'M'){
                somaHomens = somaHomens + idade;
            }

            i++;
        }

        float media = soma/5f;
        float mediaHomens = (float) somaHomens /homensCadastrados;

        System.out.println("Quantos homens foram cadastrados? " +homensCadastrados+
                           "\nQuantas mulheres foram cadastradas? " +mulheresCadastradas+
                           "\nMédia de idade do grupo: " +media+
                           "\nMédia de idade dos homens: " +mediaHomens+
                           "\nQuantas mulheres tem mais de 20 anos? " +mulheresMais20);
    }
}

File: guanabara-python/src/lista04/exercicio53.py
i = 0
homens_cadastrados = 0
mulheres_cadastradas = 0
mulheres_acima_20 = 0
soma_idade = 0
soma_idade_homens = 0

while i < 5:
    idade = int(input("Insira a idade: "))
    genero = int(input("Insira o gênero (1 - Masculino 2 - Feminino): "))

    if genero == 1:
        homens_cadastrados = homens_cadastrados + 1
        soma_idade_homens = soma_idade_homens + idade
    else:
        mulheres_cadastradas = mulheres_cadastradas + 1

    if genero == 2 and idade > 20:
        mulheres_acima_20 = mulheres_acima_20 + 1

    soma_idade = soma_idade + idade
    i = i + 1

media = soma_idade / i
media_idade_homens = soma_idade_homens / homens_cadastrados

print(f"Homens cadastrados: {homens_cadastrados}\nMulheres cadastradas: {mulheres_cadastradas}\nMédia de idade: {media}"
      f"\nMédia de idade dos homens: {media_idade_homens}\nMulheres com mais de 20 anos: {mulheres_acima_20}")
